include/sys/fs/zfs.h

```diff
@@ -261,8 +261,7 @@ typedef enum {
 	ZPOOL_NUM_PROPS
 } zpool_prop_t;
 
-/* Small enough to not hog a whole line of printout in zpool(8). */
-#define	ZPROP_MAX_COMMENT	32
+#define	ZPROP_MAX_COMMENT	8192
```

Comment:
I believe the comment is also stored in the vdev label which is only 112K in size. I don't know how much of the label is already being used but we probably want to be careful that we don't consume all the space which would then prevent storing other necessary config information in that region.

Reply:
Note that `zdb -ll` will give stats on label consumption, for example:
```ZFS Label NVList Config Stats:
  10844 bytes used, 103804 bytes free (using  9.5%)

   integers:  111   3952 bytes (36.44%)
    strings:   92   6364 bytes (58.69%)
   booleans:    2     92 bytes ( 0.85%)
    nvlists:   19    436 bytes ( 4.02%)

 leaf vdevs:   16    695 bytes average
                    1428 bytes largest
  space for 149 additional leaf vdevs
```

Reply:
Instead of bumping up the value, we have a new proposal: **save comment property in both MOS config (full string) and vdev label (first 32 characters)**

write:
- truncate comment if it is greater than 32 characters and save it to vdev label
- always save full comment to mos config

read:
- make sure spa_config reads the comment from MOS config in zpool import
- only zdb -l that would dump out the vdev label informaiton
- still document 32 character limit will continue to exist on the label, maybe introduce a new limit for the entire comment stored in mos config?

Maybe we can close this PR.